plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,270 @@
+            return bitmaps;
+        }
+        else if (storageType.equals(INLINE_MARKER) || storageType.equals(PATH_MARKER)) {
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage type for deletion vector: " + storageType);
+        }
+        throw new IllegalArgumentException("Unexpected storage type: " + storageType);
+    }
+
+    public static String toFileName(String pathOrInlineDv)
+    {
+        int randomPrefixLength = pathOrInlineDv.length() - Base85Codec.ENCODED_UUID_LENGTH;
+        String randomPrefix = pathOrInlineDv.substring(0, randomPrefixLength);
+        checkArgument(ALPHANUMERIC.matchesAllOf(randomPrefix), "Random prefix must be alphanumeric: %s", randomPrefix);
+        String prefix = randomPrefix.isEmpty() ? "" : randomPrefix + "/";
+        String encodedUuid = pathOrInlineDv.substring(randomPrefixLength);
+        UUID uuid = Base85Codec.decodeUuid(encodedUuid);
+        return "%sdeletion_vector_%s.bin".formatted(prefix, uuid);
+    }
+
+    public static ByteBuffer readDeletionVector(TrinoInputFile inputFile, int offset, int expectedSize)
+            throws IOException
+    {
+        byte[] bytes = new byte[expectedSize];
+        TrinoDataInputStream inputStream = new TrinoDataInputStream(inputFile.newStream());
+        inputStream.seek(offset);
+        int actualSize = inputStream.readInt();
+        if (actualSize != expectedSize) {
+            // TODO: Investigate why these size differ
+            log.warn("The size of deletion vector %s expects %s but got %s", inputFile.location(), expectedSize, actualSize);
+        }
+        inputStream.readFully(bytes);
+        return ByteBuffer.wrap(bytes).order(LITTLE_ENDIAN);
+    }
+
+    public static boolean contains(RoaringBitmap[] bitmaps, long value)
+    {
+        int high = highBytes(value);
+        if (high >= bitmaps.length) {
+            return false;
+        }
+        RoaringBitmap highBitmap = bitmaps[high];
+        int low = lowBytes(value);
+        return highBitmap.contains(low);
+    }
+
+    private static int highBytes(long value)
+    {
+        return toIntExact(value >> 32);
+    }
+
+    private static int lowBytes(long value)
+    {
+        return toIntExact(value);
+    }
+
+    public static RoaringBitmap[] deserializeDeletionVectors(ByteBuffer buffer)
+            throws IOException
+    {
+        checkArgument(buffer.order() == LITTLE_ENDIAN, "Byte order must be little endian: %s", buffer.order());
+        int magicNumber = buffer.getInt();
+        if (magicNumber == PORTABLE_ROARING_BITMAP_MAGIC_NUMBER) {
+            int size = toIntExact(buffer.getLong());
+            RoaringBitmap[] bitmaps = new RoaringBitmap[size];
+            for (int i = 0; i < size; i++) {
+                int key = buffer.getInt();
+                checkArgument(key >= 0);
+                RoaringBitmap bitmap = new RoaringBitmap();
+                bitmap.deserialize(buffer);
+                bitmaps[i] = bitmap;
```

Comment:
```
bitmaps[key] = bitmap;
```

the bitmaps's index is storing high 32-bits of the values  (the `key`)

however

- let's wrap `RoaringBitmap[]` in a class that will hide that from us
```
public class RoaringBitmap64
{
    public void add(int highBits, RoaringBitmap lowBits);
    
    public boolean contains(long value);
}
```

OR 

```
public class DeltaRoaringBitmap64
{
    // Deserialize Delta 64-bit extension to RoaringBitmap specification as described at
    // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#deletion-vector-format
    public static RoaringBitmap64 deserialize(byte[] bytes);
    
    public boolean contains(long value);
}
```

- we should not assume values are more or less consecutive. internally we should use a map (hashmap will do, no need for it to be sorted)

Reply:
Changed to `Roaring64NavigableMap` instead.